src/commands/data/schemas/util/metaschema.json

```diff
@@ -232,24 +233,214 @@
-              "document"
-            ],
+        "anyOf": [
+          { "$ref": "#/definitions/inputCondition" },
+          { "$ref": "#/definitions/documentCondition" },
+          { "$ref": "#/definitions/initiatorHasRelationToUserInDataCondition" },
+          { "$ref": "#/definitions/initiatorHasRelationToGroupInData" }
+        ]
+      }
+    },
+    "inputCondition": {
+      "type": "object",
+      "properties": {
+        "configuration": { "$ref": "#/definitions/configurations" },
+        "type": {
+          "type": "string",
+          "enum": ["input"]
+        },
+        "required": {
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      }
+    },
+    "documentCondition": {
+      "type": "object",
+      "properties": {
+        "configuration": { "$ref": "#/definitions/configurations" },
+        "type": {
+          "type": "string",
+          "enum": ["input"]
+        },
+        "required": {
+          "type": "array",
+          "items": { "type": "string" }
+        }
+      }
+    },
+    "initiatorHasRelationToUserInDataCondition": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "initiatorHasRelationToUserInData"
+          ]
+        },
+        "userIdField": {
+          "type": "string"
+        },
+        "relation": {
+          "type": "string",
+          "enum": ["isStaffOfTargetPatient"]
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "initiatorHasRelationToGroupInData": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "initiatorHasRelationToGroupInData"
+          ]
+        },
+        "groupIdField": {
+          "type": "string"
+        },
+        "relation": {
+          "type": "string",
+          "enum": ["staff", "patient"]
+        },
+        "requiredPermission": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "objectConfiguration": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": ["object"]
+        },
+        "required": {
+          "type": "array",
+          "items": {
             "type": "string"
           }
         },
-        "required": [
-          "type"
-        ]
+        "queryable": {
+          "type": "boolean"
+        }
+      },
+      "additionalProperties": true,
+      "required": ["type"]
+    },
+    "arrayConfiguration": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": ["array"]
+        },
+        "items": {
+          "type": "object"
+        },
+        "minItems": {
+          "type": "number"
+        },
+        "maxItems": {
+          "type": "number"
+        },
+        "contains": {
```

Comment:
Would be more correct, but this file is used by ajv to validate user input. Internally, Avj compiles this schema. If we set a reference to `"#/definitions/configurations"`, it fails to compile because of an unending loop. It should be possible to support loops in the schema, by providing a custom compiler. My knowledge of Avj is very basic, so would require additional research on my end. However, I believe this would increase the scope of the ticket to much

Reply:
Ahh weird. The `properties` are also validated and have the same schema. But I see that is an external schema, so maybe that helps with regard to being recursive 🤔 But also fine for me to keep `{ type: 'object' }` for now then. Would create another ticket to tackle that at some point then

Reply:
https://extrahorizon.atlassian.net/browse/SVCS-1989